Dockerfile

```diff
@@ -2,7 +2,8 @@ FROM node:12 as frontend-builder
 
 WORKDIR /frontend
 COPY package.json package-lock.json /frontend/
-RUN npm ci
+COPY viz-lib /frontend/viz-lib
+RUN npm ci --unsafe-perm
```

Comment:
This seems to be required to run the preinstall script (which installs viz-lib deps and build it) when we run the command as root user. An alternative is to switch the user to the redash one in this scope. I tried to switch to it for all commands, but didn't seem that quick. Another option is to switch to the user only while running this command.

Reply:
> but didn't seem that quick

in terms of effort required to make the switch?

Regardless, `--unsafe-perm` should be safe in the context of running inside a Docker container build...

Reply:
> in terms of effort required to make the switch?

In terms of changes in that file, mostly needs to update chown/chmod permissions for that folder. As the `--unsafe-perm` seemed simpler, I wanted to show it as an option, but needed to confirm regarding safety. I actually didn't sit down and actually tried to change thos yesterday, but I did today and I could get it to work like this:

```Dockerfile
FROM node:12 as frontend-builder

RUN useradd --create-home redash

WORKDIR /frontend

COPY package.json package-lock.json webpack.config.js /frontend/
COPY viz-lib /frontend/viz-lib
COPY client /frontend/client

RUN chown -R redash /frontend
USER redash
RUN npm ci
RUN ls -l /frontend
RUN npm run build
```

(notice I changed the `COPY` order to run only one `chown`, not sure if there was a reason to have it after the `npm ci` command though)

Anyway, LMK which option do you prefer.

Reply:
> (notice I changed the COPY order to run only one chown, not sure if there was a reason to have it after the npm ci command though)

The reason for the current order is to make sure that we don't have to reinstall if `package.json`/`package-lock.json` didn't change, as they change less frequently than the codebase.

Let's go with the unsafe option for now.